src/hash/rpo/tests.rs

```diff
@@ -51,6 +52,22 @@ fn hash_elements_vs_merge() {
     assert_eq!(m_result, h_result);
 }
 
+#[test]
+fn hash_elements_vs_merge_with_domain() {
+    let elements = [Felt::new(rand_value()); 8];
+
+    let digests: [RpoDigest; 2] = [
+        RpoDigest::new(elements[..4].try_into().unwrap()),
+        RpoDigest::new(elements[4..].try_into().unwrap()),
+    ];
+
+    let domain = Felt::new(rand_value());
+
+    let m_result = Rpo256::merge_in_domain(&digests, domain);
+    let h_result = hash_elements_with_domain(&elements, domain);
+    assert_eq!(m_result, h_result);
```

Comment:
Instead of adding this new helper, can we build the 12-element `state` manually in this test (with the domain element in the correct place), then apply the permutation to `state`, then `assert_eq!(m_result, state[DIGEST_RANGE]`? I think it would be a cleaner test

Reply:
Yes, I've moved the helper method logic to the `hash_elements_vs_merge_in_domain` unit tests.